model/infinispan/src/main/java/org/keycloak/models/cache/infinispan/UserCacheSession.java

```diff
@@ -705,7 +729,7 @@ public Stream<UserConsentModel> getConsentsStream(RealmModel realm, String userI
         if (cached == null) {
             Long loaded = cache.getCurrentRevision(cacheKey);
             List<UserConsentModel> consents = getDelegate().getConsentsStream(realm, userId).collect(Collectors.toList());
-            cached = new CachedUserConsents(loaded, cacheKey, realm, consents);
+            cached = new CachedUserConsents(loaded, cacheKey, realm, consents.stream().map(CachedUserConsent::new).collect(Collectors.toList()));
```

Comment:
I believe that this `getConsentsStream` cannot rely on cached value anymore since that is filled gradually, and thus should always defer to delegate.
This should not represent a performance issue since there are not many places where all consents are listed.

Reply:
I'm not changing how we are caching all user consent. Or do you mean something else?

Reply:
Sorry, I see what you mean now.

Reply:
I'm not sure if we should just call the delegate. It might impact use cases where users are querying their consents (e.g.: account console) multiple times.

I've updated the PR to call the delegate to try to fix a test failure. But I'm OK to provide a solution to keep caching all user consents.

wdyt?

Reply:
@hmlnarik Instead of returning the delegate I'm keeping the cache for all user consent. For that, the cached entry is invalidated if it was previously partially cached (e.g.: calling getConsentByClient).

The behavior should be:

* If all user consents are cached, the cache is hot, and both `getConsentsStream` and `getConsentByClient` operate on the cached data.
* If user consent is cached via `getConsentByClient`, whenever you try to fetch all user consents the cached entry is invalidated and replaced with all user consents.

Reply:
Right, that works. Thanks for fixing it!